packages/carbon-graphs/src/js/core/BaseConfig/helper.js

```diff
@@ -57,15 +63,15 @@ export const getType = (type) => getDefaultValue(type, AXIS_TYPE.DEFAULT);
  * @private
  * @returns {string} Clip path ID
  */
-export const generateClipPathId = () => `carbon-${+new Date()}-clip`;
+export const generateClipPathId = () => `carbon-${+new Date()}-${idRef()}-clip`;
 
 /**
  * Generates a clip path ID for Dateline based on current date
  *
  * @private
  * @returns {string} Clip path Dateline ID
  */
-export const generateDatelineClipPathId = () => `carbon-${+new Date()}-dateline-clip`;
+export const generateDatelineClipPathId = () => `carbon-${+new Date()}-${idRef()}-dateline-clip`;
```

Comment:
hmmmm, just curious, what exactly is this fixing? at a glance, it looks like the ids are already unique: `-clip` vs `dateline-clip`. is there a collision outside of these two ids?

Reply:
Well, id is unique in all the cases but one of our consumers is mocking the date which results in different graphs creating the same id. So taking that case into consideration we have to add extra unique things to make ID unique.

Reply:
but wouldn't the ids still be unique in that case? `${mock date}-clip` !== `${mock date}-dateline-clip`. not trying to be a stickler, just trying to understand the bug.

i agree that uuids should be used to we can guarantee that ids are unique when someone renders multiple graphs.

Reply:
When we generate two graphs in same page 1st graph will graph will have ${mock date}-clip !== ${mock date}-dateline-clip and second graph will also consists of ${mock date}-clip !== ${mock date}-dateline-clip. In that case, when we try to remove content from the first graph the content from the second graph also gets deleted since they both consist of the same id. And Yes we can remove the date.

**Note**: Problem exists when there are two graphs in same page.

Reply:
Removed date from clip-path id [here](https://github.com/cerner/terra-graphs/pull/117/commits/316365904951708854fec20afc1a374298693967#diff-606396e5aa13b62e41638acfeb1622b804a2c42a86dcf491ae3c52c60fd49389L66).